src/sync_pre_commit_lock/config.py

```diff
@@ -17,6 +18,22 @@
     pass
 
 
+def env_as_bool(value: str) -> bool:
+    return (value or "False").lower() in ("true", "1")
+
+
+def env_as_list(value: str) -> list[str]:
+    return [v.strip() for v in (value or "").split(",")]
+
+
+ENV_PREFIX = "SYNC_PRE_COMMIT_LOCK"
+ENV_CAST = {
```

Comment:
Instead of an isolated mapping, could we consider adding the cast methods in the dataclass fields, maybe in metadata `env_cast`?

Reply:
I took the liberty to use a `TypedDict` to describe the used metadata (`Required` and `NotRequired` being Python 3.11+, I use `total=False`)